src/OrchardCore.Modules/OrchardCore.Lists/Drivers/ContainedPartDisplayDriver.cs

```diff
@@ -84,5 +84,78 @@ public override async Task<IDisplayResult> UpdateAsync(ContentItem model, IUpdat
 
             return await EditAsync(model, updater);
         }
+
+        private IDisplayResult BuildViewModel(string containerId, string containerContentType, string contentType, bool enableOrdering = false)
+        {
+            var results = new List<IDisplayResult>()
+            {
+                Initialize<EditContainedPartViewModel>("ListPart_ContainerId", m =>
+                {
+                    m.ContainerId = containerId;
+                    m.ContainerContentType = containerContentType;
+                    m.EnableOrdering = enableOrdering;
+                    m.ContentType = contentType;
+                })
+                .Location("Content")
+            };
+
+            if (!String.IsNullOrEmpty(containerContentType))
+            {
+                var definition = _contentDefinitionManager.GetTypeDefinition(containerContentType);
+
+                if (definition != null)
+                {
+                    var listPart = definition.Parts.FirstOrDefault(x => x.PartDefinition.Name == nameof(ListPart));
+                    var settings = listPart?.GetSettings<ListPartSettings>();
+
+                    if (settings != null)
+                    {
+                        // Add list part navigation
+                        results.Add(Initialize<ListPartNavigationViewModel>("ListPartNavigation", async model =>
+                        {
+                            model.ContainedContentTypeDefinitions = GetContainedContentTypes(settings).ToArray();
+                            model.Container = await _contentManager.GetAsync(containerId);
+                            model.EnableOrdering = settings.EnableOrdering;
+                            model.ContainerContentTypeDefinition = definition;
+                        }).Location("Content:1.5"));
+
+                        if (settings.ShowHeader)
+                        {
+                            results.Add(GetListPartHeader(containerId, settings));
+                        }
+                    }
+                }
+            }
+
+            return Combine(results);
+        }
+
+        private IDisplayResult GetListPartHeader(string containerId, ListPartSettings listPartSettings)
+        {
+            return Initialize<ListPartHeaderViewModel>("ListPartHeader", async model =>
+            {
+                var container = await _contentManager.GetAsync(containerId, VersionOptions.Published);
```

Comment:
And here if you really want the Published you can omit the `VersionOptions.Published` which is the default.